index.bs

```diff
@@ -16,13 +16,443 @@ WPT Display: closed
+A [=request=] has a boolean [=struct/item=] <dfn for="request">eligible for storage-access</dfn>. This indicates whether the relevant third-party site allows the relevant "<a permission><code>storage-access</code></a>" permission grant to be used to allow access to cookies, if it exists. Unless stated otherwise it is false.
+
+NOTE: The [=user agent=] is expected to read a [=request=]'s [=request/eligible for storage-access=] boolean when attaching cookies to an outgoing [=request=]. It is not yet possible to specify this, since the cookie store has no defined interface that specs can interact with.
+
+NOTE: A [=request=] also has a [=request/client=], which is an [=environment settings object=] (a kind of [=environment=]) and therefore has a [=environment/has storage access=] boolean. That value is distinct from the value of the [=request=]'s [=request/eligible for storage-access=] boolean.
+
+A [=request=] has an associated <dfn for="request">single-hop cache mode</dfn>, whose value is null or a [=request/cache mode=]. 
+
+This document renames a [=request=]'s [=request/cache mode=] field to <dfn for="request">internal cache mode</dfn>.
+
+<div algorithm="request cache mode">
+This document redefines a [=request=]'s <dfn for="request" export>cache mode</dfn> as the cache mode returned by running the following steps, given a [=request=] |request|:
+    1. If |request|'s [=request/single-hop cache mode=] is not null, return |request|'s [=request/single-hop cache mode=].
+    1. Return |request|'s [=request/internal cache mode=].
+</div>
+
+A <dfn>storage access status</dfn> is one of "`none`", "`inactive`", or "`active`".
+
+<div algorithm>
+    The <dfn for=request>storage access status</dfn> of a [=request=] is the [=storage access status=], or null, returned by running the following steps:
+        1. If the user agent's cookie store would attach cookies with the `SameSite=Strict` attribute to |request|, return null. [[!COOKIES]]
+        1. Let |allowed| be a [=boolean=], initially set to the result of determining whether the user agent's cookie store allows unpartitioned cookies to be attached to |request|.
+        1. If |allowed| is true, return "`active`".
+        1. If |request|'s [=request/eligible for storage-access=] is false, then:
+            1. Set |request|'s [=request/eligible for storage-access=] to true.
+            1. Set |allowed| to be the result of determining whether the user agent's cookie store allows unpartitioned cookies to be attached to |request|.
+
+            NOTE: The cookie store is expected to [=get the current permission state=] given "`storage-access`" and |request|'s [=request/client=], and to examine |request|'s [=request/eligible for storage-access=] field and |request|'s [=request/client=]'s [=environment/has storage access=] field, when running the above step.
+
+            1. Set |request|'s [=request/eligible for storage-access=] to false.
+            1. If |allowed| is true, return "`inactive`".
```

Comment:
I think it would be good to describe in a note somewhere here, under what circumstance it is expected that setting "eligible for storage-access" to true and re-running the "determining whether the user agent's cookie store allows unpartitioned cookies to be attached to |request|" (as we do above) would set `|allowed|` to true, and under what circumstance it would not be enough to set `|allowed|` to true, and keep it false.

At least this is not obvious to me at all.